src/index.js

```diff
@@ -1,6 +1,40 @@
 import Joi from "@hapi/joi";
 export { default as Joi } from "@hapi/joi";
 
+const alternativeValidation = async (alternate, schema, values, config) => {
+  let errorMsg = "";
+  let _original = {};
+  let details = [];
+
+  for (const v in values) {
+    try {
+      await handleErrorWithSource(
+        alternate,
+        Joi.alternatives()
+          .try({
+            ...schema
+          })
+          .validateAsync(values[v], {
+            allowUnknown: !config.denyUnknown.includes(v)
+          })
+      );
+      return; // returning after first validation succeed
+    } catch (err) {
+      errorMsg += `${err.message}; `;
```

Comment:
why we need to concatenate the error message? could this be a misleading error message at the end?

Reply:
I'm giving information about all failed errors. For example why body failed and why query failed

Reply:
got it.